src/Container.php

```diff
@@ -161,50 +162,38 @@ private function resolveParams(string $qn, ReflectionMethod $constructor): itera
                 yield $param->getDefaultValue();
                 continue;
             }
+
             $paramType = $param->getType();
+
             if ($paramType instanceof ReflectionUnionType) {
-                $msgFormat = 'Unable to resolve %s constructor parameter $%s (position %d): it has union type %s.';
-                $types = [];
-                foreach ($paramType->getTypes() as $namedType) {
-                    $types[] = $namedType->getName();
-                }
-                $message = sprintf(
-                    $msgFormat,
-                    $qn,
-                    $param->getName(),
-                    $param->getPosition() + 1,
-                    implode('|', $types)
-                );
-                throw new ContainerException($this->buildExceptionReport($message));
+                $this->throwNonResolvableUnionParameterTypeException($qn, $param);
```

Comment:
Instead of this if for `ReflectionUnionType` and `throwNonResolvableUnionParameterTypeException()` call inside, we can create something like $this->checkForUnionType() that will do both